packages/pluggableWidgets/accordion-web/src/Accordion.tsx

```diff
@@ -0,0 +1,36 @@
+import { createElement, ReactElement, useMemo } from "react";
+
+import AccordionComponent from "./components/Accordion";
+
+import { AccordionContainerProps } from "../typings/AccordionProps";
+import { AccGroup } from "./components/AccordionGroup";
+
+export function Accordion(props: AccordionContainerProps): ReactElement | null {
+    const accordionGroups: AccGroup[] | undefined = useMemo(() => {
+        if (props.groups.some(group => group.visible.value === undefined || group.headerText.value === undefined)) {
+            return undefined;
```

Comment:
Is this per design? Because It feels weird that if one according group is undefined, we just stop rendering all together. Wouldn't just filtering it out make more sense? Or am I missing something?

Reply:
I think the first time the widget becomes visible it should always be fully loaded. That's how it's done in many other widgets as well. Anyhow, I will reach out to Chris to verify whether this is the desired behaviour. If he wants to change the behaviour, I will create a separate story for it. So you may resolve this conversation 😬

Reply:
Correct me if I'm wrong, but the conditional indicates to me that "if any group visibility is not defined or has no headertext, then we decide to not show any group", which doesn't sound too "it should always be fully loaded". Or am I missing something?

Reply:
I am not following you fully here. Both values will be `undefined` in the case that the data isn't available yet, which can only happen when mounting the widget. Once all data has been loaded and calculated in the client, we want to render. After that, the values changes will be applied immediately, because they will never be `undefined` anymore.

Reply:
Discussed this with Isa. We need some more information from Jelmer whether there are realistic cases where either of these values can become `undefined` during runtime (when the dynamic value turns unavailable). Because if so, the accordion widget will just stop rendering which I assume is not what we want. I will approve meanwhile but it is an improvement if it is not a super edge case

Reply:
If the accordion widget is inside a data container that is not guaranteed to always have an object (for example, if it's listening to a data grid which doesn't always have a selected object), then the visibility expressions and header texts can become unavailable, but only if they depend on the unavailable object. Standard behaviour in Mendix is to be invisible if the visibility expression cannot be evaluated, but that could be different for each individual group. I think it would make more sense to simply filter on the truthiness of the visibility expression when deciding which groups to render (and render nothing only if no groups are visible at all).

I'm not sure what should happen if the visibility expression evaluates to `true` and the header text is unavailable, though. It currently renders no groups at all if any header text is unavailable, even if the `headerRenderMode` is not text. My suggestion would be to always render the group if the visibility expression says that it's visible, as it's always possible to configure a fallback text for the header if it's unavailable (and in non-text mode, you should probably render the `headerContent` anyway).

Reply:
Created a [Jira issue](https://mendix.atlassian.net/browse/WC-508) to fix this.